src/state/slices/preferencesSlice/preferencesSlice.ts

```diff
@@ -1,22 +1,31 @@
 import { createSlice } from '@reduxjs/toolkit'
 import { ChainTypes, UtxoAccountType } from '@shapeshiftoss/types'
 
-export const accountTypePrefix = 'accountType_'
+export type Preferences = {
+  accountTypes: Record<string, any>
+}
 
-const initialState = {
-  [accountTypePrefix + ChainTypes.Bitcoin]: UtxoAccountType.SegwitP2sh
-} as { [key: string]: any }
+export const supportedAccountTypes = {
+  [ChainTypes.Bitcoin]: [
+    UtxoAccountType.SegwitNative,
+    UtxoAccountType.SegwitP2sh,
+    UtxoAccountType.P2pkh
+  ],
+  [ChainTypes.Ethereum]: undefined
+}
+
+const initialState: Preferences = {
+  accountTypes: {
+    [ChainTypes.Bitcoin]: UtxoAccountType.SegwitP2sh
+  }
+}
 
 export const preferences = createSlice({
   name: 'preferences',
   initialState,
   reducers: {
-    setPreference(state, { payload }) {
-      state[payload.key] = payload.value
+    setAccountType(state, { payload }: { payload: { key: ChainTypes; value: any } }) {
+      state.accountTypes[payload.key] = payload.value
```

Comment:
This is precisely what we were trying to avoid by having a generic key/value preference store.

The way I interpret the desired pattern to be with this is that ever type of preference we want to store has a specific reducer and state management functions to accommodate it.

Do you think that we're better off to have a strongly type Preferences that we have to update for each type of preference vs. a simple key/value store where the value is interpreted by the consumer?

Reply:
I agree this could use more discussion around how we want to more strongly type this state slice. I didn't change the functionality or typing from the original implementation, but just added the `accountTypes` preference key instead of prefixing the keys and adding a reducer to set the account type directly. This removed the need for the `getAccountTypeKey` function being passed around everywhere.